src/main/java/seedu/bookmark/model/ModelManager.java

```diff
@@ -142,9 +173,62 @@ public void updateFilteredBookList(Predicate<Book> predicate) {
 
     @Override
     public void sortFilteredBookList(Comparator<Book> comparator) {
+        if (comparator instanceof BookNameComparator) {
+            setSortingPreference(PREFIX_NAME.toString());
```

Comment:
Should we be doing these checks here? An approach worth thinking about could be to get  `SortCommandParser` to create each `SortCommand`s that know what prefix it's sorting by. Then in `SortCommand#execute()` we can just call `setSortingPreference(PREFIX_XXX)`

Reply:
Thanks for the suggestions Le Yang, I agree with you and I have made the necessary changes 😄